src/include/duckdb/common/bitpacking.hpp

```diff
@@ -112,6 +112,34 @@ class BitpackingPrimitives {
 		return num_to_round + BITPACKING_ALGORITHM_GROUP_SIZE - NumericCast<idx_t>(remainder);
 	}
 
+	inline static void BitPackBooleans(data_ptr_t dst, const data_ptr_t src, const idx_t count,
+	                                   const ValidityMask &validity_mask) {
+		bool is_last_bit_true = false; // If first value is null, write false, as it's probably the most common value
+		for (idx_t i = 0; i < count; i++) {
```

Comment:
I think we can predicate a lot from this loop (make it branchless)
updating `is_last_bit_true` can be outside of the if/else tree for starters
Maybe have a look at some of the bitpacking internals to see how they handle the bitpacking logic, I imagine that's very well optimized